src/program.js

```diff
@@ -84,8 +90,9 @@ export class Program {
   async execute(
     absolutePackageDir: string,
     {
-      systemProcess = process, logStream = defaultLogStream,
-      getVersion = defaultVersionGetter, shouldExitProgram = true,
+      checkForUpdates = defaultUpdateChecker, systemProcess = process,
+      logStream = defaultLogStream, getVersion = defaultVersionGetter,
+      shouldExitProgram = true, localEnv = 'development',
```

Comment:
This won't work in production :)

We have a [magic build variable](https://github.com/mozilla/web-ext/blob/master/webpack.config.js#L64) which is a bit confusing. You would use it like this:

    localEnv = WEBEXT_BUILD_ENV

This will set `localEnv` to the value of `NODE_ENV` at the time that `web-ext` was built.

Are you testing your patch manually? Automated tests are great but you also need to test it manually. The way I was doing it was temporarily decrementing the version to `1.5.0` in `package.json` (just for testing), running `NODE_ENV=production web-ext build` to get a build, then trying `web-ext run` or whatever to see the update notification.